### PR TITLE
Test/add specs

### PR DESCRIPTION
**Pull request by bfricka**
A few things to note about this PR:

1. `initialState` function for features. This was only implemented for root previously, but I added it to features as well. Unfortunately, the implementation is a bit different b/c features aren't using a token to provide initial state.

2. Removed `OpaqueToken`. The deps for this project are clear, and we're already using `InjectionToken` elsewhere, so it doesn't make sense to continue using a deprecated symbol.

3. Stronger types on core actions `INIT`, `UPDATE`, etc. I don't think this actually affects anything currently, but it does make certain inspections nicer.

4. Watch mode using chokidar. This should make writing development a bit nicer (`yarn run watch:tests`). This also outputs an html coverage report if that's your cup.

Some of these tests are a bit obtuse. I'm still getting used to writing ng2+ tests, so feedback is welcome if you see something that could be improved. I'll write some more tests in the future, but I wanted to get this out there.

*ps: Crossing my fingers that the `xit` test passes. Last time it failed in CI, and I have zero clue why. If it fails I'll xit back again*

### CODE REVIEW

**Comment by bfricka** [comment]
Hey there! First off, congrats on 4.0.0!

Second, I resolved conflicts!

File: modules/store/spec/store.spec.ts [hunk 1]
```diff
@@ -114,13 +108,13 @@ describe('ngRx Store', () => {
       );
     });
 
-    it('should correctly lift itself', function() {
+    it('should correctly lift itself', () => {
       const result = store.select('counter1');
 
-      expect(result instanceof Store).toBe(true);
+      expect(result).toEqual(any(Store));
```
Comment (brandonroberts) [inline modules/store/spec/store.spec.ts]: Using `instanceof` seems like a better check than `any(Store)`

File: modules/store/spec/store.spec.ts [hunk 2]
```diff
@@ -183,7 +177,7 @@ describe('ngRx Store', () => {
     });
 
     // TODO: Investigate why this is no longer working
-    xit('should complete if the dispatcher is destroyed', () => {
```
Comment (brandonroberts) [inline modules/store/spec/store.spec.ts]: Let this test remain excluded

File: modules/store/spec/utils.spec.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+/**
```
Comment (brandonroberts) [inline modules/store/spec/utils.spec.ts]: Remove header text

File: modules/store/src/store_module.ts [hunk 4]
```diff
@@ -51,7 +51,13 @@ export class StoreFeatureModule implements OnDestroy {
     @Inject(STORE_FEATURES) private features: StoreFeature<any, any>[],
     private reducerManager: ReducerManager
   ) {
-    features.forEach(feature => reducerManager.addFeature(feature));
+    for (let feature of features) {
```
Comment (brandonroberts) [inline modules/store/src/store_module.ts]: The functional approach is preferred over the for loop.

File: modules/store/src/utils.ts [hunk 5]
```diff
@@ -45,9 +45,15 @@ export function omit<T extends { [key: string]: any }>(
   object: T,
   keyToRemove: keyof T
 ): Partial<T> {
-  return Object.keys(object)
-    .filter(key => key !== keyToRemove)
-    .reduce((result, key) => Object.assign(result, { [key]: object[key] }), {});
+  const ret: Partial<T> = {};
```
Comment (brandonroberts) [inline modules/store/src/utils.ts]: No need to refactor this.

File: modules/store/src/state.ts [hunk 6]
```diff
@@ -4,20 +4,20 @@ import { Observable } from 'rxjs/Observable';
...
 import { observeOn } from 'rxjs/operator/observeOn';
-import { startWith } from 'rxjs/operator/startWith';
 import { withLatestFrom } from 'rxjs/operator/withLatestFrom';
 import { scan } from 'rxjs/operator/scan';
 import { ActionsSubject } from './actions_subject';
 import { Action, ActionReducer } from './models';
 import { INITIAL_STATE } from './tokens';
 import { ReducerObservable } from './reducer_manager';
 import { ScannedActionsSubject } from './scanned_actions_subject';
+import { INIT } from './';
```
Comment (brandonroberts) [inline modules/store/src/state.ts]: Add `INIT` to existing imports from `./actions_subject`

File: modules/store/spec/utils.spec.ts [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+ */
+
+import { omit } from '../src/utils';
+import { combineReducers, compose } from '@ngrx/store';
+
+describe(`Store utils`, () => {
+  describe(`combineReducers()`, () => {
+    const s1 = { x: '' };
+    const s2 = { y: '' };
+    const r1 = (s = s1, a: any): typeof s1 =>
```
Comment (brandonroberts) [inline modules/store/spec/utils.spec.ts]: Just a preference, but make the variable names more descriptive than `s` and `a` so they are more readable at glance

File: modules/store/src/state.ts (hunk 6 above)
Comment (bfricka) [inline modules/store/src/state.ts]: I'm sorry I don't know what you mean.

Comment (brandonroberts) [inline modules/store/src/state.ts]: On line 9, add INIT to the existing object

```ts
import { ActionsSubject, INIT } from './actions_subject';
```

**Comment by bfricka** [comment]
Not sure why Circle failed. Everything runs fine here.

File: modules/store/src/state.ts (hunk 6 above)
Comment (bfricka) [inline modules/store/src/state.ts]: Haha, got it. It took me an embarrassingly long time to figure out what you were referring to.

File: modules/store/src/store_module.ts (hunk 4 above)
Comment (bfricka) [inline modules/store/src/store_module.ts]: You got it!

**Comment by brandonroberts** [comment]
LGTM

**Pull request merged**